addons/crm_iap_mine/models/crm_lead.py

```diff
@@ -11,3 +11,16 @@ class Lead(models.Model):
 
     def _merge_get_fields(self):
         return super(Lead, self)._merge_get_fields() + ['lead_mining_request_id']
+
+    def action_generate_leads(self):
+        context = {"is_modal": True}
+        if "default_type" in self.env.context:
```

Comment:
What is the purpose of this code actually ?

Reply:
```js
export function useGenerateLeadsButton() {
    const component = useComponent();
    const user = useService("user");
    const action = useService("action");

    onWillStart(async () => {
        component.isSalesManager = await user.hasGroup("sales_team.group_sale_manager");
    });

    component.onClickGenerateLead = () => {
        const leadType = component.props.context.default_type;
        action.doAction({
            name: "Generate Leads",
            type: "ir.actions.act_window",
            res_model: "crm.iap.lead.mining.request",
            target: "new",
            views: [[false, "form"]],
            context: { is_modal: true, default_lead_type: leadType },
        });
    };
}
```
It s coming from this code. Could I remove it ?

Reply:
This code comes from the conversion of the code added by this commit https://github.com/odoo/odoo/commit/7410de111fe33731ba79f68f53e13dd3d511fb43.
To be honest, I don't really know your flows so I can test and assume that everything is going well. Wouldn't it be better if the check was done on your side ? I am only adapting the existing code.

Reply:
@FrancoisGe 
the `default_type` context key comes from crm xml views (and is correct, since the field is indeed called `type` on the model) - it's used to distinguish between leads and opportunties (depending on a setting, crm.lead records can "begin their life" with different type values by default)

@tde-banana-odoo suggests changing that, and given the code it probably does not work anyway (nothing reads this context key in a grep at least); and I believe as he seems to think that it's just not needed (as long as the context is propagated properly, the default_type key should be enough)

the if can be removed and i'll check in testing if it's ok

Reply:
> @FrancoisGe the `default_type` context key comes from crm xml views (and is correct, since the field is indeed called `type` on the model) - it's used to distinguish between leads and opportunties (depending on a setting, crm.lead records can "begin their life" with different type values by default)
> 
> @tde-banana-odoo suggests changing that, and given the code it probably does not work anyway (nothing reads this context key in a grep at least); and I believe as he seems to think that it's just not needed (as long as the context is propagated properly, the default_type key should be enough)
> 
> the if can be removed and i'll check in testing if it's ok

Ok I remove it. Could you test it ?

Reply:
iap reveal still correctly creates lead or opportunities according to the settings and the current action; so all good there AFAICT